[visionOS] requestFullscreen() may show the wrong panoramic image

yoelhawa opened the pull request:
#### d53fd76c465a8b68ef0948cebd391670c887e3f4
<pre>
[visionOS] requestFullscreen() may show the wrong panoramic image
<a href="https://bugs.webkit.org/show_bug.cgi?id=294091">https://bugs.webkit.org/show_bug.cgi?id=294091</a>
&lt;<a href="https://rdar.apple.com/152184462">rdar://152184462</a>&gt;

Reviewed by Andy Estes.

Fixes 2 bugs: ensure that image sources changes are only sent to Quick
Look if that image is in fullscreen, and ensure that the largest source
continues to get selected for fullscreen when delegating to Quick Look.

* Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp:
(WebKit::WebFullScreenManager::updateImageSource):
Stop sending image source updates to Quick Look if the image whose
source changes isn&apos;t the one in fullscreen.
(WebKit::WebFullScreenManager::enterFullScreenForElement):
(WebKit::WebFullScreenManager::didEnterFullScreen):
Move logic to force larger image sources from
WebFullScreenManager::didEnterFullScreen to
WebFullScreenManager::enterFullScreenForElement, since the former may
not be run in time anymore.

Canonical link: <a href="https://commits.webkit.org/295937@main">https://commits.webkit.org/295937@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5cb6bf3cf512098b48d4755398245f2b1a242421

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/106483 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/26232 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/16630 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/111686 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/57079 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/26899 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/34735 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/80869 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/109487 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/21331 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/96083 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/61201 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/20793 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/56520 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/90662 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/14219 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/114545 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/33621 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/24829 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/89938 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/33985 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/92314 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/89646 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22904 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/34539 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/12357 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/29230 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/33546 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/38958 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/33292 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/36645 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/34890 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

etiennesegonzac reviewed (commented):
Looking good!
@aestes or @jernoble for the official approval?

Commit:
[visionOS] requestFullscreen() may show the wrong panoramic image
https://bugs.webkit.org/show_bug.cgi?id=294091
<rdar://152184462>

Reviewed by Andy Estes.

Fixes 2 bugs: ensure that image sources changes are only sent to Quick
Look if that image is in fullscreen, and ensure that the largest source
continues to get selected for fullscreen when delegating to Quick Look.

* Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp:
(WebKit::WebFullScreenManager::updateImageSource):
Stop sending image source updates to Quick Look if the image whose
source changes isn't the one in fullscreen.
(WebKit::WebFullScreenManager::enterFullScreenForElement):
(WebKit::WebFullScreenManager::didEnterFullScreen):
Move logic to force larger image sources from
WebFullScreenManager::didEnterFullScreen to
WebFullScreenManager::enterFullScreenForElement, since the former may
not be run in time anymore.

Canonical link: https://commits.webkit.org/295937@main
